File: src/Server/WebService/ClientApp/src/app/services/net/authorization-api.ts
import { HttpClient } from "@angular/common/http";
import { Inject } from "@angular/core";
import { Observable } from "rxjs";
import { ChangePasswordParameters } from "../../model/change-password-parameters";
import { ForgotPasswordParameters } from "../../model/forgot-password-parameters";
import { LoginParameters } from "../../model/login-parameters";
import { RegisterParameters } from "../../model/register-parameters";
import { ResetPasswordParameters } from "../../model/reset-password-parameters";
import { UserInfoItem } from "../../model/userinfo-item";
import { IAuthorizationApi } from "./api/authorization-api.interface";

export class AuthorizationApi implements IAuthorizationApi {

    constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string)
    {
//        super();
    }

    RegisterAsync(registerParameters: RegisterParameters): Observable<Response> {
        throw new Error("Method not implemented.");
    }
    LoginAsync(loginParameters: LoginParameters): Observable<Response> {
        throw new Error("Method not implemented.");
    }
    LogoutAsync(): Observable<Response> {
        throw new Error("Method not implemented.");
    }
    GetUserInfo(): Observable<UserInfoItem> {
        throw new Error("Method not implemented.");
    }
    DeleteUserAsync(): Observable<boolean> {
        throw new Error("Method not implemented.");
    }
    ChangePasswordAsync(chngParameters: ChangePasswordParameters): Observable<Response> {
        throw new Error("Method not implemented.");
    }
    ForgotPasswordAsync(forgotParameters: ForgotPasswordParameters): Observable<Response> {
        throw new Error("Method not implemented.");
    }
    ResetPasswordAsync(resetPassswordParameters: ResetPasswordParameters): Observable<Response> {
        throw new Error("Method not implemented.");
    }
}

File: src/Client/Client.Net/AuthorizationApi.cs
using DevInstance.DevCoreApp.Client.Net.Api;
using DevInstance.DevCoreApp.Shared.Model;
using System.Net.Http;
using System.Net.Http.Json;
using System.Threading.Tasks;

namespace DevInstance.DevCoreApp.Client.Net;

public class AuthorizationApi : ApiBase, IAuthorizationApi
{
    private const string Controller = "api/user/account/";

    public AuthorizationApi(HttpClient http) : base (http)
    {
    }

    public async Task RegisterAsync(RegisterParameters registerParameters)
    {
        var result = await httpClient.PostAsJsonAsync(Controller + "register", registerParameters);
        result.EnsureSuccessStatusCode();
    }

    public async Task LoginAsync(LoginParameters loginParameters)
    {
        var result = await httpClient.PostAsJsonAsync(Controller + "signin", loginParameters);
        result.EnsureSuccessStatusCode();
    }

    public async Task LogoutAsync()
    {
        var result = await httpClient.PostAsync(Controller + "signout", null);
        result.EnsureSuccessStatusCode();
    }

    public async Task<UserInfoItem> GetUserInfoAsync()
    {
        return await httpClient.GetFromJsonAsync<UserInfoItem>(Controller + "user-info");
    }

    public async Task<bool> DeleteUserAsync()
    {
        var result = await httpClient.DeleteAsync($"{Controller}");
        result.EnsureSuccessStatusCode();
        return result.Content.ReadFromJsonAsync<bool>().Result;
    }

    public async Task ChangePasswordAsync(ChangePasswordParameters chngParameters)
    {
        var result = await httpClient.PutAsJsonAsync(Controller + "change-password", chngParameters);
        result.EnsureSuccessStatusCode();
    }

    public async Task ForgotPasswordAsync(ForgotPasswordParameters forgotParameters)
    {
        var result = await httpClient.PostAsJsonAsync(Controller + "forgot-password", forgotParameters);
        result.EnsureSuccessStatusCode();
    }

    public async Task ResetPasswordAsync(ResetPasswordParameters resetPassswordParameters)
    {
        var result = await httpClient.PostAsJsonAsync(Controller + "reset-password", resetPassswordParameters);
        result.EnsureSuccessStatusCode();
    }
}
